src/Avalonia.Base/Media/TransformGroup.cs

```diff
@@ -44,14 +48,16 @@ public override Matrix Value
         {
             get
             {
-                Matrix result = Matrix.Identity;
-
-                foreach (var t in Children)
+                if (_lastMatrix is null)
```

Comment:
It would add a bigger overhead when TransformGroup has a single child. As we would end up copying this struct a lot.
But I would assume it's very uncommon scenario for TransformGroup.